android/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java

```diff
@@ -116,28 +130,63 @@ public void setAnimation(final PluginCall call) {
         call.resolve();
     }
 
-    private void setupSafeAreaInsets(boolean hasFixedWebView, boolean hasMetaViewportCover) {
-        ViewCompat.setOnApplyWindowInsetsListener((View) getBridge().getWebView().getParent(), (v, insets) -> {
-            if (hasFixedWebView && hasMetaViewportCover) {
-                return insets;
-            }
+    @JavascriptInterface
+    public void onDOMReady() {
+        getActivity().runOnUiThread(() -> {
+            this.bridge.getWebView().evaluateJavascript(viewportMetaJSFunction, (res) -> {
+                hasViewportCover = res.equals("true");
```

Comment:
If you're going with the strategy where the webview is responsible for telling the plugin if it has a `viewport-fit=cover` value, wouldn't it make more sense to skip the `evaluateJavascript` part and directly let the native code know what that value is?